pyhmy/validator.py

```diff
@@ -0,0 +1,651 @@
+import json
+
+from decimal import (
+    Decimal,
+    InvalidOperation
+)
+
+from .account import (
+    get_balance,
+    is_valid_address
+)
+
+from .exceptions import (
+    InvalidValidatorError,
+    RPCError,
+    RequestsError,
+    RequestsTimeoutError
+)
+
+from .numbers import (
+    convert_atto_to_one,
+    convert_one_to_atto
+)
+
+from .staking import (
+    get_all_validator_addresses,
+    get_validator_information
+)
+
+
+_default_endpoint = 'http://localhost:9500'
+_default_timeout = 30
+
+_name_char_limit = 140
+_identity_char_limit = 140
+_website_char_limit = 140
+_security_contact_char_limit = 140
+_details_char_limit = 280
+_min_required_delegation = 10000
+
+
+# TODO: Add validator transcation functions
+# TODO: Add unit testing
+class Validator:
+
+    def __init__(self, address):
+        if not is_valid_address(address):
+            raise InvalidValidatorError(1, f'{address} is not valid ONE address')
+        self._address = address
+        self._bls_keys = []
+
+        self._name = None
+        self._identity = None
+        self._website = None
+        self._details = None
+        self._security_contact = None
+
+        self._min_self_delegation = None
+        self._max_total_delegation = None
+        self._inital_delegation = None  # amount
+
+        self._rate = None
+        self._max_change_rate = None
+        self._max_rate = None
+
+    def __repr__(self):
+        return f'Validator: {self._address}'
```

Comment:
`__repr__` is meant to be an object representation where `eval` can be called on output and get a similar object back. But if that cannot be done, then a hint message must follow the form: `<SOME MSG HERE>` (angle brackets necessary). Read more [here](https://docs.python.org/3/library/functions.html#repr).